src/pyFAI/gui/pilx/MainWindow.py

```diff
@@ -118,7 +119,17 @@ def initData(self,
                  ):
 
         self._file_name = os.path.abspath(file_name)
-        self._dataset_path = dataset_path
+
+        c=0
+        for i in dataset_path[-1::-1]:
+            if i.isdigit():
+                c+=1
+            else:
+                break
+
+        _dataset_path = dataset_path[:-c]
+        path, base = posixpath.split(_dataset_path)
+        self._dataset_paths = {}
```

Comment:
This could go with the modifs l.155 to keep the relevant code together.

Reply:
Can make sense ... my idea was to keep the context manager part as small as possible, i.e. put this outside the context manager. No problem for moving it into the `with hdf5file` section.